googleapiclient/http.py

```diff
@@ -56,11 +58,12 @@
 from googleapiclient.model import JsonModel
 from oauth2client import util
 
-
 DEFAULT_CHUNK_SIZE = 512*1024
 
 MAX_URI_LENGTH = 2048
 
+_TOO_MANY_REQUESTS = 429
+
 
 def _retry_request(http, num_retries, req_type, sleep, rand, uri, method, *args,
```

Comment:
With this change this function is now nearly one hundred lines long with seventy lines of body. What options do we have for breaking it up into an assembly of code elements each of which is a manageable size?